src/psyclone/tests/psyGen_test.py

```diff
@@ -2668,17 +2692,609 @@ def test_binaryoperation_can_be_printed():
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert "Could not process " in str(error.value)
+    assert "Unrecognized attribute type " in str(error.value)
+
+    # RHS array specifications are not supported
+    reader = FortranStringReader("integer :: l1(4)")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert ("Array specifications after the variable name are not "
+            "supported.") in str(error.value)
+
+    # Initialisations are not supported
+    reader = FortranStringReader("integer :: l1 = 1")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert ("Initializations on the declaration statements are not "
+            "supported.") in str(error.value)
+
+    # Char lengths are not supported
+    # TODO: It would be simpler to do just a Specification_Part(reader) instead
+    # of parsing a full program, but fparser/169 needs to be fixed first.
+    reader = FortranStringReader("program dummy\ncharacter :: l*4"
+                                 "\nend program")
+    program = f2008_parser(reader)
+    fparser2spec = program.content[0].content[1].content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert ("Character length specifications are not "
+            "supported.") in str(error.value)
+
+
+def test_fparser2astprocessor_process_not_supported_declarations(f2008_parser):
+    '''Test that process_declarations method raises the proper errors when
+    declarations contain unsupported attributes.
+    '''
+    from fparser.common.readfortran import FortranStringReader
+    from fparser.two.Fortran2003 import Specification_Part
+    fake_parent = KernelSchedule("dummy_schedule")
+    processor = Fparser2ASTProcessor()
+
+    reader = FortranStringReader("integer, external :: arg1")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert "Could not process " in str(error.value)
+    assert ". Unrecognized attribute " in str(error.value)
+
+    reader = FortranStringReader("integer, save :: arg1")
+    fparser2spec = Specification_Part(reader).content[0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert "Could not process " in str(error.value)
+    assert ". Unrecognized attribute " in str(error.value)
+
+
+def test_fparser2astprocessor_process_declarations_intent(f2008_parser):
```

Comment:
This one also doesn't use `f2008_parser`

Reply:
Same as above.